packages/destination-actions/src/destinations/loops/createOrUpdateContact/index.ts

```diff
@@ -146,10 +122,27 @@ const action: ActionDefinition<Settings, Payload> = {
       json: {
         ...(typeof customAttributes === 'object' && customAttributes),
         ...rest,
-        mailingLists: formattedMailingLists
+        mailingLists: filteredMailingLists || {}
       }
     })
   }
 }
 
 export default action
+
+// remove any key value pairs where the value is not a true or false string
+function filterMailingLists(obj: { [key: string]: unknown }): { [key: string]: boolean } {
+  const result: { [key: string]: boolean } = {}
+  for (const key in obj) {
+    if (typeof obj[key] !== 'boolean') {
+      if (obj[key] === 'true') {
+        result[key] = true
+      } else if (obj[key] === 'false') {
+        result[key] = false
+      }
+    } else {
+      result[key] = obj[key]
+    }
+  }
```

Comment:
```suggestion
for (const key in obj) {
    const value = obj[key];
    
    if (typeof value === 'string') {
      if (value === 'true') {
        result[key] = true;
      } else if (value === 'false') {
        result[key] = false;
      }
    } else if (typeof value === 'boolean') {
      result[key] = value;
    }
  }
```

Getting a type issue here with the build, perhaps try the above^: 
<img width="1030" alt="Screenshot 2024-09-27 at 1 20 33 PM" src="https://github.com/user-attachments/assets/c0ca317d-3966-4ea5-a2bd-91dbf16bb793">